### PR TITLE
style: UI fixes to conform to Postman Design System

### LINKED ISSUE #760: Updated color codes
Color codes have been updated to pass accessibility tests. New colors have been added, and color style Primary/Coral has been changed to Error/Fire Red. Please check new color codes against existing components.

Please refer to [slide 3: Colors](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on changes. Our updated color guide can also be viewed in [figma](https://www.figma.com/file/Wd6jTqpdaqaItVLoyk1Me2/%F0%9F%8E%A8Postman-Design-System?node-id=3559%3A1).

### LINKED ISSUE #761: Updated text styles & typography fixes
Text styles have been updated to follow the 8pt grid system. Please refer to the figma file for the comprehensive guide. We've also noted some wrongly used text styles for specific copies.

Please refer to [slides 4-7: Typography](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for notes on changes. Our updated typography guidelines can also be viewed in [Figma](https://www.figma.com/file/Wd6jTqpdaqaItVLoyk1Me2/%F0%9F%8E%A8Postman-Design-System?node-id=3562%3A73).

### LINKED ISSUE #762: Button fixes
We've updated the buttons to follow the 8-pt grid system. Buttons are the wrong height, some have the wrong text style, and icons and spacing are the wrong size. 

Please refer to [slide 8-15: buttons](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on changes. Our updated button guide can also be viewed in [Figma](https://www.figma.com/file/Wd6jTqpdaqaItVLoyk1Me2/%F0%9F%8E%A8Postman-Design-System?node-id=3573%3A260).

### LINKED ISSUE #764: Campaign Table fixes
Please refer to [slide 22-25: Campaign Table](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on fixes.

### LINKED ISSUE #765: Progress Table fixes
Please refer to [slide 26-27: Progress Table](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on fixes.

### LINKED ISSUE #766: Sidebar fixes
Please refer to [slide 28: Sidebar](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on fixes.

### LINKED ISSUE #767: Toast fixes
Please refer to [slide 29: Toasts](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on fixes.

### LINKED ISSUE #768: Pagination fixes
Please refer to [slide 30: Pagination](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on fixes.

### LINKED ISSUE #770: Creation Screens layout fixes
Please refer to [slide 33-35: Creation screens](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on fixes.

### LINKED ISSUE #771: Modal fixes
Please refer to [slide 36: Modals](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing) for more notes on fixes.

### PR DESCRIPTION
**Pull request by lamkeewei**
Using this PR to consolidate all the UI fixes required. 

Enabled Amplify build for branch and hosted at https://uifixes.postman.gov.sg for easy review by designers.

Related #800 #803 

Closes #760 
Closes #761 
Closes #762 
Closes #764 
Closes #765  
Closes #766 
Closes #767 
Closes #768 
Closes #770 
Closes #771 

## Solution

**Improvements**:
- Consolidate variables and typography based on Postman Design System
- Split up different styles (colours, spacings, typography) into separate files for better organisation
- Introduce more semantic SCSS variable names (`primary`, `primary-light`, etc) to make it easier to consistently apply the design system. 
- Class/variable names tries to follow as closely as possible to those used in Figma

## References
- [Postman Design System](https://www.figma.com/file/Wd6jTqpdaqaItVLoyk1Me2/%F0%9F%8E%A8Postman-Design-System?node-id=3559%3A1)
- [UI review slides](https://docs.google.com/presentation/d/1gk4npC6zRzw3OQjykm1QB5loQ0v-x2q4tRxBcuJ29tg/edit?usp=sharing)
- Postman Design Master (in Figma)

### CODE REVIEW

File: frontend/src/styles/base/_spacings.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Spacing scale
+$spacing-1: 0.5rem;
```
Comment (jeantanzj) [inline frontend/src/styles/base/_spacings.scss:2]: Curious, where did you get these values?

Comment (lamkeewei) [reply frontend/src/styles/base/_spacings.scss:2]: It's in the design system. The designers defined some fixed set of spacing for laying out elements. 

image

Comment (jeantanzj) [reply frontend/src/styles/base/_spacings.scss:2]: Thanks for pointing this out, I didn't scroll to the right on figma.

File: frontend/src/styles/base/_colours.scss [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+$error: $red;
+$error-light: $red-100;
+$error-dark: $red-900;
+
+$warning: $yellow;
+$warning-light: $yellow-100;
+$warning-dark: $yellow-900;
+
+$neutral: $grey;
+$neutral-light: $grey-300;
```
Comment (jeantanzj) [inline frontend/src/styles/base/_colours.scss:60]: With reference to how material ui determines the palette's scheme,  might it be useful to have $primary-text or $secondary-text variables indicating the color to use for text in the foreground  as well?  https://material-ui.com/customization/palette/

Comment (lamkeewei) [reply frontend/src/styles/base/_colours.scss:60]: Just looked through the design system and it appears most of our text are of the same colour (`$secondary`), with the exception of each of the toasts which use the success, error and warning palettes respectively. Not too sure if the current design system has the concept of primary and secondary text colours.

**Review by jeantanzj — APPROVED** [review]

**Pull request merged**